plugins/alarmclock/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/alarmclock/AlarmClockResource.java

```diff
@@ -95,83 +84,96 @@
-		case "edit":
-			if (alarmTime == null || !alarmTime.isValid()) {
-				throw new WebApplicationException("The given time wasn't a valid time", Status.BAD_REQUEST);
-			}
-			Alarm alarm;
-			try {
-				alarm = this.logic.editAlarm(alarmNumber, alarmTime.getHour(), alarmTime.getMinute());
-			} catch (NoSuchElementException e) {
-				throw new WebApplicationException("there is no alarm" + alarmNumber, e, Status.NOT_FOUND);
-			}
-			Timestamp ts = new Timestamp(alarm);
-			ts.setLink(createAlarmPath(alarm.getId()));
-			return ts;
-		case "activate":
+	public Alarm editAlarm(@PathParam("pathid") int alarmNumber, Alarm alarmInc) {
+		Alarm alarm = alarmInc;
+		int day;
+		if (LocalDateTime.now().getDayOfMonth() == alarm.getAlarmTime().getDayOfMonth()) {
+			day = -1;
+		} else {
+			day = 1;
+		}
+		this.logic.editAlarm(alarmNumber, day, alarm.getAlarmTime().getHour(), alarm.getAlarmTime().getMinute());
+		return alarm;
+	}
+
+	/**
+	 * @param alarmNumber
+	 *            number of the alarm which should be activated or deactivated
+	 * @param alarmInc
+	 *            the incoming alarm
+	 * @return null
+	 */
+	@POST
+	@Path("alarms/de.activate/{pathid}")
+	@Produces(MediaType.APPLICATION_JSON)
+	@Consumes(MediaType.APPLICATION_JSON)
+	public Alarm activatedeactivateAlarm(@PathParam("pathid") int alarmNumber, Alarm alarmInc) {
+		Alarm alarm = alarmInc;
+		if (!alarm.isActive()) {
 			this.logic.activateAlarm(alarmNumber);
-			break;
-		case "delete":
-			this.logic.deleteAlarm(alarmNumber);
-			break;
-		case "deactivate":
+		} else if (alarm.isActive()) {
 			this.logic.deactivateAlarm(alarmNumber);
-			break;
-		default:
-			throw new WebApplicationException(500);
 		}
 		return null;
 	}
 
+	/**
+	 * @param alarmNumber
+	 *            number of the alarm which should be deleted
+	 * @return null
+	 */
+	@POST
+	@Path("alarms/delete/{pathid}")
+	@Produces(MediaType.APPLICATION_JSON)
+	@Consumes(MediaType.APPLICATION_JSON)
+	public Alarm deleteAlarm(@PathParam("pathid") int alarmNumber) {
+		this.logic.deleteAlarm(alarmNumber);
+		return null;
+	}
+
 	/**
 	 * sets a alarm to a given timestamp
 	 * 
-	 * @param alarmTime
+	 * @param alarm
 	 *            the timestamp for the alarm
 	 * @return the newly created alarm
 	 */
 	@POST
 	@Path("alarms/new")
 	@Consumes(MediaType.APPLICATION_JSON)
 	@Produces(MediaType.APPLICATION_JSON)
-	public Timestamp newAlarm(Timestamp alarmTime) {
-		if (alarmTime.isValid()) {
-			Alarm result = this.logic.setAlarm(alarmTime.getHour(), alarmTime.getMinute());
-			alarmTime.setLink(createAlarmPath(result.getId()));
-			return alarmTime;
+	public Alarm newAlarm(Alarm alarm) {
+		int day;
+		if (LocalDateTime.now().getDayOfMonth() == alarm.getAlarmTime().getDayOfMonth()) {
+			day = -1;
+		} else {
+			day = 1;
 		}
-		throw new WebApplicationException("The given time wasn't a valid time", Status.BAD_REQUEST);
+		return this.logic.setAlarm(day, alarm.getAlarmTime().getHour(), alarm.getAlarmTime().getMinute());
```

Comment:
also set a link here
catch the exception thrown by the logic and rethrow the old exception